core/dbt/contracts/graph/model_config.py

```diff
@@ -42,6 +42,20 @@ def _set_meta_value(obj: M, key: str, existing: Optional[Dict[str, Any]] = None)
     return result
 
 
+class OnConfigurationChange(str, Enum):
```

Comment:
I wonder if some naming like OnConfigurationChangeOptions might be better here? The name doesn't feel intuitive.

Reply:
kinda had the same thought for the NodeConfig option as well planned on raising it as it seems to wide range and people might not know its only for MV's